unumpy/_multimethods.py

```diff
@@ -1379,7 +1389,20 @@ def _ediff1d_argreplacer(args, kwargs, dispatchables):
     )
 
 
-@create_numpy(_ediff1d_argreplacer)
+def _ediff1d_default(ary, to_end=None, to_begin=None):
+    ary = ravel(ary)
+
+    diffs = ary[1:] - ary[:-1]
+
+    if to_end is None:
+        to_end = asarray([], dtype=ary.dtype)
```

Comment:
What's going on here? Concatenating an empty array does nothing.

Reply:
You are right, I could do the other way around like:

```python
if to_end is not None:
    ary = concatenate([ary, ravel(to_end)])
```

Seems better this way!

Reply:
Append to a list, check for trivial case and return early.